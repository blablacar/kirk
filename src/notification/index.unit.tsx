import React from 'react'
import renderer from 'react-test-renderer'
import Notification from 'notification'
import Button from 'button'
import exenv from 'exenv'

const close = jest.fn()

describe('Notification', () => {
  it('should show the notification when isOpen is true', () => {
    const wrapper = mount(<Notification isOpen close={close} />)
    expect(wrapper.find('.kirk-notification').exists()).toBe(true)
  })
  it('should not show the notification when isOpen is false', () => {
    const wrapper = mount(<Notification isOpen={false} close={close} />)
    expect(wrapper.find('.kirk-notification').exists()).toBe(false)
  })
  it('should call close method when click on cross button', () => {
    const wrapper = mount(<Notification isOpen close={close} />)
    wrapper.find(Button).simulate('click')
    expect(close).toHaveBeenCalled()
  })
  it('should render the same layout on client and server side', () => {
    exenv.canUseDOM = false
    const serverSide = mount(<Notification isOpen close={close} />)
    exenv.canUseDOM = true
    const clientSide = mount(<Notification isOpen close={close} />)

    expect(serverSide.html()).toEqual(clientSide.html())
  })
  it('should not have changed', () => {
    exenv.canUseDOM = false
    const notificationServerSide = renderer
      .create(
        <Notification isOpen close={close}>
          Oups
        </Notification>,
      )
      .toJSON()
    exenv.canUseDOM = true
    expect(notificationServerSide).toMatchSnapshot()
  })
})
