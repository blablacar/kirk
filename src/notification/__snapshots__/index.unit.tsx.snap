// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should not have changed 1`] = `
<div
  className="transition-wrapper kirk-notification-container"
>
  <div
    className="jsx-2214296095 kirk-notification slide-up slide-up-entered"
  >
    <span
      className="kirk-text kirk-text-title kirk-notification-content"
    >
      Oups
    </span>
    <button
      className="kirk-button kirk-button-unstyled kirk-notification-cross"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <span
        className="jsx-4075288211"
      >
        <svg
          aria-hidden={true}
          className="kirk-icon"
          height="18px"
          viewBox="0 0 24 24"
          width="18px"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <path
            d="M19 5L5 19M19 19L5 5"
            fill="none"
            stroke="#FFF"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
